src/wp-includes/rest-api.php

```diff
@@ -1539,6 +1539,67 @@ function rest_stabilize_value( $value ) {
 	return $value;
 }
 
+/**
+ * Gets the possible error of combining operation.
+ *
+ * @since 5.6.0
+ *
+ * @param array  $value  The value to validate.
+ * @param string $param  The parameter name, used in error messages.
+ * @param array  $errors The errors array, to search for possible error.
+ * @return WP_Error      The combining operation error.
+ */
+function rest_get_combining_operation_error( $value, $param, $errors ) {
+	// If there is only one error, simply return it.
+	if ( 1 === count( $errors ) ) {
+		$reason = $errors[0]['error_object']->get_error_message();
+
+		/* translators: 1: Parameter, 2: Reason. */
+		return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s does not match any schema. Reason: %2$s' ), $param, $reason ) );
+	}
+
+	// Filter out all errors related to type validation.
+	$filtered_errors = array();
+	foreach ( $errors as $error ) {
+		$best_type = rest_get_best_type_for_value( $value, array( $error['schema']['type'] ) );
+		if ( $best_type === $error['schema']['type'] ) {
```

Comment:
I think this doesn't account for an array of types.

Reply:
Something we could do instead, is to adjust the error code from `rest_invalid_param` to `rest_invalid_type` in `rest_validate_value_from_schema` when the value fails validation because of the type. Then we don't need to repeat the validation logic here and can instead filter out errors based on that error code.

Reply:
Could any external code use these error codes? In other words, could these changes break other code?

Reply:
Yep! My thinking is that `rest_invalid_param` is essentially the generic error code we use in that function. So narrowing the error code would improve things, but shouldn't have a negative effect in the vast majority of cases since the error code can't be used to disambiguate on its own.

This would also only be relevant for people calling the function directly, since we discard the exact error codes in `WP_REST_Request::has_valid_params()` and just use the error message.